source/docs/software/advanced-controls/trajectories/holonomic.rst

```diff
@@ -0,0 +1,86 @@
+Holonomic Drive Controller
+==========================
+The holonomic drive controller is a trajectory tracker for robots with holonomic drivetrains (i.e. swerve and mecanum). This can be used to accurately track trajectories with correction for minor disturbances.
+
+Constructing a Holonomic Drive Controller
+-----------------------------------------
+The holonomic drive controller should be instantiated with 2 PID controllers and 1 profiled PID controller.
+
+.. note:: For more information on PID control, see :ref:`docs/software/advanced-controls/controllers/pidcontroller:PID Control in WPILib`.
+
+The 2 PID controllers are controllers that should correct for error in the field-relative x and y directions respectively. For example, if the first 2 arguments are ``PIDController(1, 0, 0)`` and ``PIDController(1.2, 0, 0)`` respectively, the holonomic drive controller will add an additional meter per second in the x direction for every meter of error in the x direction and will add an additional 1.2 meters per second in the y direction for every meter of error in the y direction.
+
+The final parameter is a ``ProfiledPIDController`` for the rotation of the robot. Because the rotation dynamics of a holonomic drivetrain are decoupled from movement in the x and y directions, users can set custom rotation references while following a trajectory. These rotation references are profiled according to the parameters set in the ``ProfiledPIDController``.
```

Comment:
```suggestion
The final parameter is a ``ProfiledPIDController`` for the rotation of the robot. Because the rotation dynamics of a holonomic drivetrain are decoupled from movement in the x and y directions, users can set custom heading references while following a trajectory. These heading references are profiled according to the parameters set in the ``ProfiledPIDController``.
```